src/pages/work/[slug].tsx

```diff
@@ -32,68 +45,111 @@ export async function getStaticProps(context: { params: { slug: string } }) {
     return match ? parseInt(match[0], 10) : 0;
   }
 
-  const number = getNumberFromString(slug);
+  const id = getNumberFromString(slug);
+  const nextId = (id + 1) % images.length;
   return {
     props: {
-      data: { slug, image: images[number], number },
+      data: {
+        id,
+        slug,
+        image: images[id],
+        nextProject: {
+          id: nextId,
+          image: images[nextId],
+        },
+      },
     },
   };
 }
 
-export default function Work({
-  data,
-}: InferGetStaticPropsType<typeof getStaticProps>) {
+interface Props extends InferGetStaticPropsType<typeof getStaticProps> {}
+
+export default function Work({ data }: Props) {
   const { state } = useGlobalState();
+  const nextProjectHref = `/work/item-${data.nextProject.id}`;
+  const [imgSrc, setImgSrc] = useState(data.image);
+  const [title, setTitle] = useState(`/work/${data.slug}`);
+  const [nextRoute, setNextRoute] = useState<string | null>(null);
+  const [hideContent, setHideContent] = useState(false);
+  const [hideFooter, setHideFooter] = useState(false);
 
   async function handleClick(event: MouseEvent<HTMLAnchorElement>) {
-    const href = event.currentTarget.getAttribute("href");
+    const { currentTarget } = event;
+    const href = currentTarget.getAttribute("href");
+
+    setNextRoute(href);
 
     event.preventDefault();
 
-    await animate(document.documentElement.scrollTop, 0, {
+    const rect = event.currentTarget.getBoundingClientRect();
+    const scrollTo = rect.top + window.scrollY;
+
+    setHideContent(true);
+    /* Scroll back to top of window */
+    await animate(document.documentElement.scrollTop, scrollTo, {
       duration: 1,
       ease: [0.645, 0.045, 0.355, 1],
       onUpdate: (v) => {
         window.scrollTo(0, v);
       },
-    }).then(() => href && router.push(href));
+    })
+      .then(() => {
+        /* Hide/change things before route change */
+        setHideFooter(true);
```

Comment:
I don't think the code in this part is necessary - the footer already contains the next projects media and title, we shouldn't need to hide it or set the data  differently, just use layoutid to animate the footer to the header

Reply:
This was addressing some of the issues [I put in the ticket](https://www.notion.so/piersolenski/28c93fb7e97d4ba1a7c621cb75d5ea7f?v=17c0747b6c85427e9283863c78eb00ab&p=0ef67d6d5c614f6eb64c259a5c96c88d&pm=s). But I haven't played around with unsetting the `layoutId` as you've said below.

But ultimately the problem comes with managing when the page scrolls and the position of the footer element on the page / in the viewport - that's what manually hiding the element was tryna acheive. But it does lead to a flicker, so not ideal.